packages/resilience/src/middleware.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { createLogger } from '@sap-cloud-sdk/util/dist/logger';
+import {
+  MiddlewareContext,
+  executeWithMiddleware,
+  MiddlewareIn
+} from './middleware';
+
+describe('middleware', () => {
+  const logger = createLogger('middleware');
+
+  const middleWareAppend1 = function (
+    options: MiddlewareIn<string, string, MiddlewareContext<string>>
+  ) {
+    const id = 'append1';
+    return s => {
+      logger.info(id);
+      return options.fn(s + ' ' + id);
+    };
+  };
+  const middleWareAppend2 = function (
+    options: MiddlewareIn<string, string, MiddlewareContext<string>>
+  ) {
+    const id = 'append2';
+    return s => {
+      logger.info(id);
+      return options.fn(s + ' ' + id);
+    };
+  };
+
+  const middleWareSkip = function (
+    options: MiddlewareIn<string, string, MiddlewareContext<string>>
+  ) {
+    options.skipNext();
+    return options.fn;
+  };
+
+  const middleWareAdjustArgument = function (
+    options: MiddlewareIn<string, string, MiddlewareContext<string>>
+  ) {
+    options.context.fnArgument = 'Changed by middleware';
+    return options.fn;
+  };
+
+  it('adds middlewares in the expected order', async () => {
+    const infoSpy = jest.spyOn(logger, 'info');
+    const actual = await executeWithMiddleware(
+      [middleWareAppend1, middleWareAppend2],
+      { uri: 'dummyUri', fnArgument: 'initial Input', tenantId: 'dummyTenant' },
+      (s: string) => Promise.resolve(s)
+    );
+    expect(infoSpy).toHaveBeenNthCalledWith(2, 'append1');
+    expect(infoSpy).toHaveBeenNthCalledWith(1, 'append2');
+    expect(actual).toBe('initial Input append2 append1');
```

Comment:
[q] I am not sure I understand, but shouldn't `append1` be added first?

Reply:
The functions are like a stack so the one added first is executed last.